kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed

kv/kvserver.TestReplicateQueueDecommissioningNonVoters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7596542?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7596542?buildTab=artifacts#/) on master @ [8357abb668a5adaff781343b394b162fb1b66c6e](https://github.com/cockroachdb/cockroach/commits/8357abb668a5adaff781343b394b162fb1b66c6e):


```
=== RUN   TestReplicateQueueDecommissioningNonVoters
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters3913840834
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestReplicateQueueDecommissioningNonVoters
    replicate_queue_test.go:791: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters3913840834
--- FAIL: TestReplicateQueueDecommissioningNonVoters (212.58s)
=== RUN   TestReplicateQueueDecommissioningNonVoters/replace
    replicate_queue_test.go:664: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1772/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/replicate_queue_test.go:664
        	Error:      	Condition never satisfied
        	Test:       	TestReplicateQueueDecommissioningNonVoters/replace
    --- FAIL: TestReplicateQueueDecommissioningNonVoters/replace (107.45s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueDecommissioningNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-21632

kv/kvserver.TestReplicateQueueDecommissioningNonVoters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7837114?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7837114?buildTab=artifacts#/) on master @ [8a5cb5109d6b7e122e996a44f940bfd61698cd81](https://github.com/cockroachdb/cockroach/commits/8a5cb5109d6b7e122e996a44f940bfd61698cd81):


```
=== RUN   TestReplicateQueueDecommissioningNonVoters
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters2796954702
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestReplicateQueueDecommissioningNonVoters
    replicate_queue_test.go:791: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters2796954702
--- FAIL: TestReplicateQueueDecommissioningNonVoters (217.79s)
=== RUN   TestReplicateQueueDecommissioningNonVoters/remove
    replicate_queue_test.go:765: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1165/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/replicate_queue_test.go:765
        	Error:      	Condition never satisfied
        	Test:       	TestReplicateQueueDecommissioningNonVoters/remove
    --- FAIL: TestReplicateQueueDecommissioningNonVoters/remove (111.82s)
=== RUN   TestReplicateQueueDecommissioningNonVoters/replace
    replicate_queue_test.go:664: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1165/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/replicate_queue_test.go:664
        	Error:      	Condition never satisfied
        	Test:       	TestReplicateQueueDecommissioningNonVoters/replace
    --- FAIL: TestReplicateQueueDecommissioningNonVoters/replace (105.88s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92894 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2.0]
- #92726 kv/kvserver: TestReplicateQueueDecommissioningNonVoters/replace failed [C-test-failure O-robot T-kv branch-release-22.2]
- #92317 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueDecommissioningNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestReplicateQueueDecommissioningNonVoters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7869620?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7869620?buildTab=artifacts#/) on master @ [1b4947bf4e1036a7c55996d2cb6ab815254a3c8c](https://github.com/cockroachdb/cockroach/commits/1b4947bf4e1036a7c55996d2cb6ab815254a3c8c):


```
=== RUN   TestReplicateQueueDecommissioningNonVoters
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters2295618678
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestReplicateQueueDecommissioningNonVoters
    replicate_queue_test.go:791: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters2295618678
--- FAIL: TestReplicateQueueDecommissioningNonVoters (205.74s)
=== RUN   TestReplicateQueueDecommissioningNonVoters/replace
    replicate_queue_test.go:664: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3121/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/replicate_queue_test.go:664
        	Error:      	Condition never satisfied
        	Test:       	TestReplicateQueueDecommissioningNonVoters/replace
    --- FAIL: TestReplicateQueueDecommissioningNonVoters/replace (93.41s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92894 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2.0]
- #92726 kv/kvserver: TestReplicateQueueDecommissioningNonVoters/replace failed [C-test-failure O-robot T-kv branch-release-22.2]
- #92317 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueDecommissioningNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestReplicateQueueDecommissioningNonVoters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7939301?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7939301?buildTab=artifacts#/) on master @ [f04f44cf71df608c513073b73dd02f7cb68143e5](https://github.com/cockroachdb/cockroach/commits/f04f44cf71df608c513073b73dd02f7cb68143e5):


```
=== RUN   TestReplicateQueueDecommissioningNonVoters
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters2327261729
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestReplicateQueueDecommissioningNonVoters
    replicate_queue_test.go:791: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters2327261729
--- FAIL: TestReplicateQueueDecommissioningNonVoters (228.61s)
=== RUN   TestReplicateQueueDecommissioningNonVoters/remove
    replicate_queue_test.go:765: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2776/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/replicate_queue_test.go:765
        	Error:      	Condition never satisfied
        	Test:       	TestReplicateQueueDecommissioningNonVoters/remove
    --- FAIL: TestReplicateQueueDecommissioningNonVoters/remove (113.13s)
=== RUN   TestReplicateQueueDecommissioningNonVoters/replace
    replicate_queue_test.go:664: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2776/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/replicate_queue_test.go:664
        	Error:      	Condition never satisfied
        	Test:       	TestReplicateQueueDecommissioningNonVoters/replace
    --- FAIL: TestReplicateQueueDecommissioningNonVoters/replace (115.47s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92894 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2.0]
- #92726 kv/kvserver: TestReplicateQueueDecommissioningNonVoters/replace failed [C-test-failure O-robot T-kv branch-release-22.2]
- #92317 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueDecommissioningNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestReplicateQueueDecommissioningNonVoters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7971169?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7971169?buildTab=artifacts#/) on master @ [d6f98e90684894fd36f53596e6aac355676d232e](https://github.com/cockroachdb/cockroach/commits/d6f98e90684894fd36f53596e6aac355676d232e):


```
=== RUN   TestReplicateQueueDecommissioningNonVoters
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters307267310
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestReplicateQueueDecommissioningNonVoters
    replicate_queue_test.go:791: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters307267310
--- FAIL: TestReplicateQueueDecommissioningNonVoters (192.97s)
=== RUN   TestReplicateQueueDecommissioningNonVoters/replace
    replicate_queue_test.go:664: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2777/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/replicate_queue_test.go:664
        	Error:      	Condition never satisfied
        	Test:       	TestReplicateQueueDecommissioningNonVoters/replace
    --- FAIL: TestReplicateQueueDecommissioningNonVoters/replace (93.17s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92894 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2.0]
- #92726 kv/kvserver: TestReplicateQueueDecommissioningNonVoters/replace failed [C-test-failure O-robot T-kv branch-release-22.2]
- #92317 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueDecommissioningNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestReplicateQueueDecommissioningNonVoters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8016040?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8016040?buildTab=artifacts#/) on master @ [94e03c016955dfd64d4e15358ea92226a8f362aa](https://github.com/cockroachdb/cockroach/commits/94e03c016955dfd64d4e15358ea92226a8f362aa):


```
=== RUN   TestReplicateQueueDecommissioningNonVoters
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters244428719
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestReplicateQueueDecommissioningNonVoters
    replicate_queue_test.go:791: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters244428719
--- FAIL: TestReplicateQueueDecommissioningNonVoters (191.90s)
=== RUN   TestReplicateQueueDecommissioningNonVoters/remove
    replicate_queue_test.go:765: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3081/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/replicate_queue_test.go:765
        	Error:      	Condition never satisfied
        	Test:       	TestReplicateQueueDecommissioningNonVoters/remove
    --- FAIL: TestReplicateQueueDecommissioningNonVoters/remove (101.54s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92894 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2.0]
- #92726 kv/kvserver: TestReplicateQueueDecommissioningNonVoters/replace failed [C-test-failure O-robot T-kv branch-release-22.2]
- #92317 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueDecommissioningNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestReplicateQueueDecommissioningNonVoters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8032753?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8032753?buildTab=artifacts#/) on master @ [61f7c816574848f804f1adf7f42f7315bfcaaeba](https://github.com/cockroachdb/cockroach/commits/61f7c816574848f804f1adf7f42f7315bfcaaeba):


```
=== RUN   TestReplicateQueueDecommissioningNonVoters
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters3774653508
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestReplicateQueueDecommissioningNonVoters
    replicate_queue_test.go:791: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters3774653508
--- FAIL: TestReplicateQueueDecommissioningNonVoters (191.81s)
=== RUN   TestReplicateQueueDecommissioningNonVoters/remove
    replicate_queue_test.go:765: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2533/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/replicate_queue_test.go:765
        	Error:      	Condition never satisfied
        	Test:       	TestReplicateQueueDecommissioningNonVoters/remove
    --- FAIL: TestReplicateQueueDecommissioningNonVoters/remove (99.43s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92894 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2.0]
- #92726 kv/kvserver: TestReplicateQueueDecommissioningNonVoters/replace failed [C-test-failure O-robot T-kv branch-release-22.2]
- #92317 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueDecommissioningNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestReplicateQueueDecommissioningNonVoters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8080118?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8080118?buildTab=artifacts#/) on master @ [4232883add85a151c423c45904ac4096d04656c5](https://github.com/cockroachdb/cockroach/commits/4232883add85a151c423c45904ac4096d04656c5):


```
=== RUN   TestReplicateQueueDecommissioningNonVoters
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters3745988980
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestReplicateQueueDecommissioningNonVoters
    replicate_queue_test.go:791: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters3745988980
--- FAIL: TestReplicateQueueDecommissioningNonVoters (186.18s)
=== RUN   TestReplicateQueueDecommissioningNonVoters/remove
    replicate_queue_test.go:765: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2489/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/replicate_queue_test.go:765
        	Error:      	Condition never satisfied
        	Test:       	TestReplicateQueueDecommissioningNonVoters/remove
    --- FAIL: TestReplicateQueueDecommissioningNonVoters/remove (102.64s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92894 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2.0]
- #92726 kv/kvserver: TestReplicateQueueDecommissioningNonVoters/replace failed [C-test-failure O-robot T-kv branch-release-22.2]
- #92317 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueDecommissioningNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestReplicateQueueDecommissioningNonVoters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8089247?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8089247?buildTab=artifacts#/) on master @ [c4bde8b72cdd4016845ae70ef5162b3f11fab1fb](https://github.com/cockroachdb/cockroach/commits/c4bde8b72cdd4016845ae70ef5162b3f11fab1fb):


```
=== RUN   TestReplicateQueueDecommissioningNonVoters
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters4274010653
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestReplicateQueueDecommissioningNonVoters
    replicate_queue_test.go:791: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters4274010653
--- FAIL: TestReplicateQueueDecommissioningNonVoters (197.70s)
=== RUN   TestReplicateQueueDecommissioningNonVoters/remove
    replicate_queue_test.go:765: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2717/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/replicate_queue_test.go:765
        	Error:      	Condition never satisfied
        	Test:       	TestReplicateQueueDecommissioningNonVoters/remove
    --- FAIL: TestReplicateQueueDecommissioningNonVoters/remove (99.41s)
=== RUN   TestReplicateQueueDecommissioningNonVoters/replace
    replicate_queue_test.go:664: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2717/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/replicate_queue_test.go:664
        	Error:      	Condition never satisfied
        	Test:       	TestReplicateQueueDecommissioningNonVoters/replace
    --- FAIL: TestReplicateQueueDecommissioningNonVoters/replace (98.28s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92894 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2.0]
- #92726 kv/kvserver: TestReplicateQueueDecommissioningNonVoters/replace failed [C-test-failure O-robot T-kv branch-release-22.2]
- #92317 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueDecommissioningNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestReplicateQueueDecommissioningNonVoters [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8129170?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8129170?buildTab=artifacts#/) on master @ [64e4fc9faa4e0ab19fe5ba78f053bc2b1390cb5e](https://github.com/cockroachdb/cockroach/commits/64e4fc9faa4e0ab19fe5ba78f053bc2b1390cb5e):


```
=== RUN   TestReplicateQueueDecommissioningNonVoters
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters2192878514
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestReplicateQueueDecommissioningNonVoters
    replicate_queue_test.go:791: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueDecommissioningNonVoters2192878514
--- FAIL: TestReplicateQueueDecommissioningNonVoters (183.67s)
=== RUN   TestReplicateQueueDecommissioningNonVoters/remove
    replicate_queue_test.go:765: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1160/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_test_/kvserver_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/replicate_queue_test.go:765
        	Error:      	Condition never satisfied
        	Test:       	TestReplicateQueueDecommissioningNonVoters/remove
    --- FAIL: TestReplicateQueueDecommissioningNonVoters/remove (101.07s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92894 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2.0]
- #92726 kv/kvserver: TestReplicateQueueDecommissioningNonVoters/replace failed [C-test-failure O-robot T-kv branch-release-22.2]
- #92317 kv/kvserver: TestReplicateQueueDecommissioningNonVoters failed [C-test-failure O-robot T-kv branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueDecommissioningNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -594,6 +594,7 @@ func TestReplicateQueueDecommissioningNonVoters(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 	skip.UnderRace(t, "takes a long time or times out under race")
+	skip.UnderDeadlockWithIssue(t, 94383)
 
 	ctx := context.Background()
 

